src/utils/parse.ts

```diff
@@ -85,18 +86,13 @@ export function parse(
   // https://github.com/eslint/eslint/blob/3ec436ee/lib/linter.js#L637
   parserOptions.filePath = path
 
+  parserOptions = withoutProjectParserOptions(parserOptions) as TSESLint.ParserOptions
+
```

Comment:
@JoshuaKGoldberg It seems that there is a type mismatch here. The return type of `withoutProjectParserOptions` is `TSESTree.TSESTreeOptions` and can not be assigned to `TSESLint.ParserOptions`.

Reply:
> Filed https://github.com/typescript-eslint/typescript-eslint/issues/9827.

Nice!

> We could take an issue suggesting that if you want to advocate for it.

It is not a hard requirement for me as well. Since `@typescript-eslint/estree` is already a transitive dependency (through `@typescript-eslint/utils`) of `eslint-plugin-import-x`, explicitly adding `@typescript-eslint/estree` won't actually introducing a new package, so I am OK with that.